lib/rules/no-magic-numbers.js

```diff
@@ -26,10 +26,129 @@ function normalizeIgnoreValue(x) {
 	return x;
 }
 
+/**
+ * Gets the true parent of the literal, handling prefixed numbers (-1 / +1)
+ * @param {ASTNode} node The literal node
+ * @returns {ASTNode} The parent node
+ */
+function getLiteralParent(node) {
+	if (["-", "+"].includes(node.parent.operator)) {
+		return node.parent.parent;
+	}
+
+	return node.parent;
+}
+
+/**
+ * Checks if the node parent is a TypeScript enum member
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if the node parent is a TypeScript enum member
+ */
+function isParentTSEnumDeclaration(node) {
+	const parent = getLiteralParent(node);
+	return parent.type === "TSEnumMember";
+}
+
+/**
+ * Checks if any ancestor of the node is a TypeScript type alias declaration
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if any ancestor is a TypeScript type alias declaration
+ */
+function isInsideTSTypeAliasDeclaration(node) {
+	let current = node;
+
+	while (current && current.parent) {
+		if (current.parent.type === "TSTypeAliasDeclaration") {
+			return true;
+		}
+		current = current.parent;
+	}
+
+	return false;
+}
+
+/**
+ * Checks if the node grandparent is a TypeScript union type and its parent is a type alias declaration
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if conditions are met
+ */
+function isGrandparentTSUnionType(node) {
+	if (node.parent.parent.type === "TSUnionType") {
+		return isInsideTSTypeAliasDeclaration(node.parent.parent);
+	}
+
+	return false;
+}
+
+/**
+ * Checks if the node is a valid TypeScript numeric literal type.
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if the node is a TypeScript numeric literal type
+ */
+function isTSNumericLiteralType(node) {
+	let nodeToCheck = node;
+	// For negative numbers, use the parent node
+	if (node.parent.operator === "-") {
+		nodeToCheck = node.parent;
+	}
+
+	// Use the more thorough check for deeply nested types
+	if (isInsideTSTypeAliasDeclaration(nodeToCheck)) {
+		return true;
+	}
+
+	// If the grandparent is a TSUnionType and it's parent is a TSTypeAliasDeclaration, ignore
+	if (isGrandparentTSUnionType(nodeToCheck)) {
+		return true;
+	}
```

Comment:
This doesn't seem to have any effect, i.e., when I remove it, all the tests are still passing. Because `isInsideTSTypeAliasDeclaration(nodeToCheck)` above already checks if there's `TSTypeAliasDeclaration` anywhere in the ancestry.

Reply:
Fixed.